### PR TITLE
Small improvements to blobstorage binding

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
Spin-off from PR adding contexts to input bindings

- Adds timeout to container creation to avoid hanging
- Simpler/faster parsing of metadata
- A couple more smaller things

### CODE REVIEW

File: bindings/azure/blobstorage/blobstorage.go [hunk 1]
```diff
@@ -153,14 +156,8 @@ func (a *AzureBlobStorage) Init(metadata bindings.Metadata) error {
...
 func (a *AzureBlobStorage) parseMetadata(metadata bindings.Metadata) (*blobStorageMetadata, error) {
-	connInfo := metadata.Properties
-	b, err := json.Marshal(connInfo)
-	if err != nil {
-		return nil, err
-	}
-
 	var m blobStorageMetadata
-	err = json.Unmarshal(b, &m)
+	err := mapstructure.WeakDecode(metadata.Properties, &m)
```
Comment (berndverst) [inline bindings/azure/blobstorage/blobstorage.go:160]: Why is this necessary?

Comment (ItalyPaleAle) [reply bindings/azure/blobstorage/blobstorage.go:160]: Not strictly necessary, but avoids marshalling to JSON and back. It's just more efficient (and the module was already imported in Dapr).

**Review by berndverst — APPROVED** [review]
